drivers/adc/ad463x/ad463x.c

```diff
@@ -314,26 +350,50 @@ static int32_t ad463x_init_gpio(struct ad463x_dev *dev,
 				struct ad463x_init_param *init_param)
 {
 	int32_t ret;
-
+	/* configure reset pin */
 	ret = no_os_gpio_get_optional(&dev->gpio_resetn, init_param->gpio_resetn);
 	if (ret != 0)
 		return ret;
 
 	/** Reset to configure pins */
-	if (dev->gpio_resetn) {
-		ret = no_os_gpio_direction_output(dev->gpio_resetn, NO_OS_GPIO_LOW);
-		if (ret != 0)
-			return ret;
+	ret = no_os_gpio_direction_output(dev->gpio_resetn, NO_OS_GPIO_LOW);
+	if (ret != 0)
+		goto error_gpio_reset;
```

Comment:
the previous implementation looks correct to me.
resetn is not allocated in this function so it should not be removed here, simply return without calling the remove function,

Reply:
Isn't resetn allocated in `no_os_gpio_get_optional` ?  `no_os_gpio_direction_output` and `no_os_gpio_set_value` only fails when allocated, right? 
To summarize: the gpio optional is to keep the driver working even when this GPIOs are not set (in this case nothing will be allocate and it won't return error). but once it is set and allocated, it need to be removed after an error. correct me if I am wrong or if you meant something else.

Reply:
for some reason I thought dev->gpio_resetn is allocated in ad463x_init (because it is already removed there)...
but I see that it is allocated in  ad463x_init_gpio, so it is alright. I deleted all my comments related to this.